test/boost/commitlog_test.cc

```diff
@@ -1091,3 +1092,136 @@ BOOST_AUTO_TEST_CASE(test_commitlog_segment_descriptor) {
     }
 }
 
+static future<> do_test_commitlog_replay_compression(std::string_view compressor_name) {
+    auto cfg = make_shared<db::config>();
+    tmpdir tmp;
+
+    // set commitlog compression as well as a cross-restart persistent data dir
+    cfg->commitlog_compression({{ "class_name", compressor_name.data() }});
+    cfg->data_file_directories({tmp.path().string()});
+
+    co_await do_with_cql_env_thread([] (cql_test_env& env) {
+        env.execute_cql("create table t (pk text primary key, v text)").get();
+        env.execute_cql("insert into ks.t (pk, v) values ('apa', 'ko')").get();
+
+        // fake unclean shutdown by simply dropping all commitlog segments.
+        env.db().invoke_on_all([](auto& db) -> future<> {
+            co_await db.commitlog()->sync_all_segments();
+            co_await db.commitlog()->release();
+        }).get();
+
+    }, cfg);
+
+    // remove all sstables for our keyspace so we can test for data being
+    // recovered by commitlog
+    fs::remove_all(tmp.path() / "ks");
```

Comment:
This should be a no-op, yes? since we didn't flush any memtables.

Reply:
It should, but since a flush can happen during runtime as well (though probably not, since we don't particularly fill the memtables), it is prudent to do.

Reply:
Agree